src/lock.rs

```diff
@@ -205,28 +225,44 @@ impl LockedPackage {
     }
 
     /// Validates if another LockedPackage matches this one
-    pub fn validate(&self, package: &Package) -> eyre::Result<()> {
+    pub fn validate(&self, package: &Package) -> miette::Result<()> {
         let digest: Digest = digest::digest(&digest::SHA256, &package.tgz)
             .try_into()
             .unwrap();
 
+        #[derive(Error, Debug)]
+        #[error("{property} mismatch - expected {expected}, actual {actual}")]
+        struct ValidationError {
+            property: &'static str,
+            expected: String,
```

Comment:
can errors have lifetimes?

Reply:
Not sure what you mean, errors are just normal types, so I guess yes.